core/src/main/java/org/sonarsource/sonarlint/core/rules/CleanCodePrinciples.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.rules;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import javax.annotation.CheckForNull;
+import org.apache.commons.io.IOUtils;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+
+class CleanCodePrinciples {
+
+  @CheckForNull
+  public static String getContent(String key) {
```

Comment:
@mithil-dobarkar-sonarsource in the end Damien moved content to resources. Maybe you can discuss with him the rational to see if you should do the same on your side for the "other" context content.

Reply:
My rationale is that it is static content, it was a bit ugly in the code (we don't have text blocks) and I had a lot of issues from Sonar :)
In this case I think it is more logical to have them as resources